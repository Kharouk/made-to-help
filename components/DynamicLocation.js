import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { MapView, Location, Permissions } from 'expo';
import MapViewDirections from 'react-native-maps-directions';
import KEY from '../env.config';
import DisplayAlertMarkers from './DisplayAlertMarkers'

const GEOLOCATION_OPTIONS = { enableHighAccuracy: true };


const origin = '50 commercial st UK';
const destination = 'Westminster Bridge, London SE1 7GP';

const styles = StyleSheet.create({
  calloutView: {
    backgroundColor: 'rgba(255, 255, 255, 0.3)',
    width: 500,
  },
  calloutText: {
    fontSize: 40,
    color: 'hotpink',
    textAlign: 'center',
    fontWeight: 'bold',
    fontFamily: 'Verdana',
  },
});

const DB_URL = KEY;

export default class DynamicLocation extends React.Component {
  state = {
    /* eslint-disable-next-line */
    location: { coords: { latitude: -0.09, longitude: 51 } },
    journeyDistance: null,
    journeyTime: null,
    isLoading: true,
    markers: [],
  };

  componentDidMount = async () => {
    await Permissions.askAsync(Permissions.LOCATION);
    Location.watchPositionAsync(GEOLOCATION_OPTIONS, this.locationChanged);
    this.fetchMarkerData();
  };

  /* eslint-disable */
  locationChanged = location => {
    region = {
      latitude: location.coords.latitude,
      longitude: location.coords.longitude,
      latitudeDelta: 0.1,
      longitudeDelta: 0.05,
    };
    this.setState({ location, region });
  };
  /* eslint-enable */

  fetchMarkerData = () => {
    fetch('https://made-to-help.herokuapp.com/api/stations')
      .then(response => response.json())
      .then((responseJson) => {
        this.setState({ isLoading: false, markers: responseJson });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    return (
      <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'center' }}>
        <MapView
          style={{ flex: 1 }}
          showsUserLocation
          region={this.state.region}
          provider="google"
        >
          {this.state.isLoading ? null
            : this.state.markers.map((marker, index) => {
              const coords = {
                latitude: marker.latitude,
                longitude: marker.longitude,
              };
              const stepData = `Accessibility: ${marker.stepFree}`;

              return (
                <MapView.Marker
                  key={index}
                  coordinate={coords}
                  title={marker.station}
                  description={stepData}
                  image={require('../assets/images/wheelchair-access.png')}
                />
              );
            })}
          <MapViewDirections
            origin={origin}
            destination={destination}
            apikey={DB_URL}
            strokeWidth={3}
            strokeColor="hotpink"
            mode="walking"
            onReady={(result) => {
              this.setState({
                journeyTime: result.duration,
                journeyDistance: result.distance.toFixed(2),
              });
            }}
          />
        </MapView>
        <MapView.Callout>
          <View style={styles.calloutView}>
            <Text
              style={styles.calloutText}
              accessibilityLabel={`Hello! Journey Time is ${Math.round(this.state.journeyTime)} Minutes and the distance is \n ${this.state.journeyDistance} KM`}
            >
              {`${Math.round(this.state.journeyTime)} Minutes \n ${this.state.journeyDistance} KM`}
            </Text>
          </View>
          <DisplayAlertMarkers />
        </MapView.Callout>
      </View>
    );
  }
}
