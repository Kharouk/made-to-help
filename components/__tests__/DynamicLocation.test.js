import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DynamicLocation from '../DynamicLocation';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-native-maps', () => {
  const React = require('react');
  return class MockMapView extends React.Component {
    static Marker = props => React.createElement('Marker', props, props.children);

    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  };
});

test('Renders DynamicLocation component', () => {
  const snapshot = renderer.create(<DynamicLocation />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

it('Should test for setState changes to locationChanged', () => {
  const snapshot = renderer.create(<DynamicLocation />);

  const instance = snapshot.getInstance()
  expect(instance.state).toMatchSnapshot('something');

  instance.locationChanged({coords: { latitude: -0.09, longitude: 51 } })
  expect(instance.state).toMatchSnapshot('Something Updated')
});

it('Should test for setState changes to journeyTime and journeyDistance', () => {
  const wrap = shallow(<DynamicLocation />);
  expect(wrap).toMatchSnapshot();
});
