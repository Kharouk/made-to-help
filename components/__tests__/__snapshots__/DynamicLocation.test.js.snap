// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders DynamicLocation component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <MapView
    provider="google"
    showsUserLocation={true}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Callout>
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "width": 500,
        }
      }
    >
      <Text
        accessibilityLabel="Hello! Journey Time is 0 Minutes and the distance is 
 null KM"
        style={
          Object {
            "color": "hotpink",
            "fontFamily": "Verdana",
            "fontSize": 40,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        0 Minutes 
 null KM
      </Text>
    </View>
  </Callout>
</View>
`;

exports[`Should test for setState changes to journeyTime and journeyDistance 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DynamicLocation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MockMapView
          provider="google"
          showsUserLocation={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <MapViewDirections
            apikey="AIzaSyCbhVNmiXfsH9OAomH1G2je9QfFqD8sqtc"
            destination="Westminster Bridge, London SE1 7GP"
            mode="walking"
            onReady={[Function]}
            origin="50 commercial st UK"
            strokeColor="hotpink"
            strokeWidth={3}
          />
        </MockMapView>,
        <Unknown>
          <Component
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.3)",
                "width": 500,
              }
            }
          >
            <Component
              accessibilityLabel="Hello! Journey Time is 0 Minutes and the distance is 
 null KM"
              style={
                Object {
                  "color": "hotpink",
                  "fontFamily": "Verdana",
                  "fontSize": 40,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              0 Minutes 
 null KM
            </Component>
          </Component>
        </Unknown>,
      ],
      "style": Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            "null - Time 
 null - Distance",
          ],
          "style": Object {
            "flex": 0.25,
          },
        },
        "ref": null,
        "rendered": Array [
          " ",
          "null - Time 
 null - Distance",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            null,
            <MapViewDirections
              apikey="AIzaSyCbhVNmiXfsH9OAomH1G2je9QfFqD8sqtc"
              destination={
                Object {
                  "latitude": 51.523018,
                  "longitude": -0.087029,
                }
              }
              mode="walking"
              onReady={[Function]}
              origin={
                Object {
                  "latitude": 51.5002,
                  "longitude": 0.1332,
                }
              }
              strokeColor="hotpink"
              strokeWidth={3}
            />,
          ],
          "provider": "google",
          "region": undefined,
          "showsUserLocation": true,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "apikey": "AIzaSyCbhVNmiXfsH9OAomH1G2je9QfFqD8sqtc",
              "destination": Object {
                "latitude": 51.523018,
                "longitude": -0.087029,
              },
              "mode": "walking",
              "onReady": [Function],
              "origin": Object {
                "latitude": 51.5002,
                "longitude": 0.1332,
              },
              "strokeColor": "hotpink",
              "strokeWidth": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Component
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.3)",
                "width": 500,
              }
            }
          >
            <Component
              accessibilityLabel="Hello! Journey Time is 0 Minutes and the distance is 
 null KM"
              style={
                Object {
                  "color": "hotpink",
                  "fontFamily": "Verdana",
                  "fontSize": 40,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              0 Minutes 
 null KM
            </Component>
          </Component>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Component
              accessibilityLabel="Hello! Journey Time is 0 Minutes and the distance is 
 null KM"
              style={
                Object {
                  "color": "hotpink",
                  "fontFamily": "Verdana",
                  "fontSize": 40,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              0 Minutes 
 null KM
            </Component>,
            "style": Object {
              "backgroundColor": "rgba(255, 255, 255, 0.3)",
              "width": 500,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessibilityLabel": "Hello! Journey Time is 0 Minutes and the distance is 
 null KM",
              "children": "0 Minutes 
 null KM",
              "style": Object {
                "color": "hotpink",
                "fontFamily": "Verdana",
                "fontSize": 40,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": "0 Minutes 
 null KM",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MockMapView
            provider="google"
            showsUserLocation={true}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <MapViewDirections
              apikey="AIzaSyCbhVNmiXfsH9OAomH1G2je9QfFqD8sqtc"
              destination="Westminster Bridge, London SE1 7GP"
              mode="walking"
              onReady={[Function]}
              origin="50 commercial st UK"
              strokeColor="hotpink"
              strokeWidth={3}
            />
          </MockMapView>,
          <Unknown>
            <Component
              style={
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0.3)",
                  "width": 500,
                }
              }
            >
              <Component
                accessibilityLabel="Hello! Journey Time is 0 Minutes and the distance is 
 null KM"
                style={
                  Object {
                    "color": "hotpink",
                    "fontFamily": "Verdana",
                    "fontSize": 40,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                0 Minutes 
 null KM
              </Component>
            </Component>
          </Unknown>,
        ],
        "style": Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              "null - Time 
 null - Distance",
            ],
            "style": Object {
              "flex": 0.25,
            },
          },
          "ref": null,
          "rendered": Array [
            " ",
            "null - Time 
 null - Distance",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              null,
              <MapViewDirections
                apikey="AIzaSyCbhVNmiXfsH9OAomH1G2je9QfFqD8sqtc"
                destination={
                  Object {
                    "latitude": 51.523018,
                    "longitude": -0.087029,
                  }
                }
                mode="walking"
                onReady={[Function]}
                origin={
                  Object {
                    "latitude": 51.5002,
                    "longitude": 0.1332,
                  }
                }
                strokeColor="hotpink"
                strokeWidth={3}
              />,
            ],
            "provider": "google",
            "region": undefined,
            "showsUserLocation": true,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "apikey": "AIzaSyCbhVNmiXfsH9OAomH1G2je9QfFqD8sqtc",
              "destination": "Westminster Bridge, London SE1 7GP",
              "mode": "walking",
              "onReady": [Function],
              "origin": "50 commercial st UK",
              "strokeColor": "hotpink",
              "strokeWidth": 3,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "apikey": "AIzaSyCbhVNmiXfsH9OAomH1G2je9QfFqD8sqtc",
                "destination": Object {
                  "latitude": 51.523018,
                  "longitude": -0.087029,
                },
                "mode": "walking",
                "onReady": [Function],
                "origin": Object {
                  "latitude": 51.5002,
                  "longitude": 0.1332,
                },
                "strokeColor": "hotpink",
                "strokeWidth": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Component
              style={
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0.3)",
                  "width": 500,
                }
              }
            >
              <Component
                accessibilityLabel="Hello! Journey Time is 0 Minutes and the distance is 
 null KM"
                style={
                  Object {
                    "color": "hotpink",
                    "fontFamily": "Verdana",
                    "fontSize": 40,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                0 Minutes 
 null KM
              </Component>
            </Component>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Component
                accessibilityLabel="Hello! Journey Time is 0 Minutes and the distance is 
 null KM"
                style={
                  Object {
                    "color": "hotpink",
                    "fontFamily": "Verdana",
                    "fontSize": 40,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                0 Minutes 
 null KM
              </Component>,
              "style": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.3)",
                "width": 500,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessibilityLabel": "Hello! Journey Time is 0 Minutes and the distance is 
 null KM",
                "children": "0 Minutes 
 null KM",
                "style": Object {
                  "color": "hotpink",
                  "fontFamily": "Verdana",
                  "fontSize": 40,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "0 Minutes 
 null KM",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Should test for setState changes to locationChanged: Something Updated 1`] = `
Object {
  "isLoading": true,
  "journeyDistance": null,
  "journeyTime": null,
  "location": Object {
    "coords": Object {
      "latitude": -0.09,
      "longitude": 51,
    },
  },
  "markers": Array [],
  "region": Object {
    "latitude": -0.09,
    "latitudeDelta": 0.1,
    "longitude": 51,
    "longitudeDelta": 0.05,
  },
}
`;

exports[`Should test for setState changes to locationChanged: something 1`] = `
Object {
  "isLoading": true,
  "journeyDistance": null,
  "journeyTime": null,
  "location": Object {
    "coords": Object {
      "latitude": -0.09,
      "longitude": 51,
    },
  },
  "markers": Array [],
}
`;
